airflow/providers/hashicorp/hooks/vault.py

```diff
@@ -0,0 +1,546 @@
+            kv_engine_version=kv_engine_version,
+            token=token,
+            username=self.connection.login,
+            password=self.connection.password,
+            secret_id=self.connection.password,
+            role_id=role_id,
+            kubernetes_role=kubernetes_role,
+            kubernetes_jwt_path=kubernetes_jwt_path,
+            gcp_key_path=gcp_key_path,
+            gcp_scopes=gcp_scopes,
+        )
+
+    def _get_kubernetes_parameters_from_connection(
+        self, kubernetes_jwt_path: Optional[str], kubernetes_role: Optional[str]) \
+            -> Tuple[str, Optional[str]]:
+        if not kubernetes_jwt_path:
+            kubernetes_jwt_path = self.connection.extra_dejson.get("kubernetes_jwt_path")
+            if not kubernetes_jwt_path:
+                kubernetes_jwt_path = DEFAULT_KUBERNETES_JWT_PATH
+        if not kubernetes_role:
+            kubernetes_role = self.connection.extra_dejson.get("kubernetes_role")
+        return kubernetes_jwt_path, kubernetes_role
+
+    def _get_gcp_parameters_from_connection(
+        self, gcp_key_path: Optional[str], gcp_scopes: Optional[str]) \
+            -> Tuple[Optional[str], Optional[str]]:
+        if not gcp_scopes:
+            gcp_scopes = self.connection.extra_dejson.get("gcp_scopes")
+        if not gcp_key_path:
+            gcp_key_path = self.connection.extra_dejson.get("gcp_key_path")
+        return gcp_key_path, gcp_scopes
+
+    def get_conn(self) -> hvac.Client:
+        """
+        Retrieves connection to Vault.
+
+        :rtype: hvac.Client
+        :return: connection used.
+        """
+        return self.vault_client.client
+
+    def get_secret(self, secret_path: str, secret_version: Optional[int] = None) -> Optional[dict]:
+        """
+        Get secret value from the engine.
+
+        :param secret_path: Path of the secret
+        :type secret_path: str
+        :param secret_version: Optional version of key to read - can only be used in case of version 2 of KV
+        :type secret_version: int
+
+        See https://hvac.readthedocs.io/en/stable/usage/secrets_engines/kv_v1.html
+        and https://hvac.readthedocs.io/en/stable/usage/secrets_engines/kv_v2.html for details.
+
+        :param secret_path: Path of the secret
+        :type secret_path: str
+        :rtype: dict
+        :return: secret stored in the vault as a dictionary
+        """
+        return self.vault_client.get_secret(secret_path=secret_path, secret_version=secret_version)
+
+    def get_secret_metadata(self, secret_path: str) -> Optional[dict]:
+        """
+        Reads secret metadata (including versions) from the engine. It is only valid for KV version 2.
+
+        :param secret_path: Path to read from
+        :type secret_path: str
+        :rtype: dict
+        :return: secret metadata. This is a Dict containing metadata for the secret.
+
+                 See https://hvac.readthedocs.io/en/stable/usage/secrets_engines/kv_v2.html for details.
```

Comment:
How does this line render in the docs?

Reply:
I fixed those and will take a look at the docs generated to confirm all is ok.